docs/src/real_world.md

```diff
@@ -529,6 +529,92 @@ julia> expensive_computation(5)
 2021-05-19 19:20:29 | n = 5 | error = 0.15 | maxrss = 326.27 MiB
 ```
 
+## Saving plots
+At this stage after you have generated data and used [`collect_results`](@ref) to gather your data. Now you may be starting to make plots/figures and wanting to save them, then you can just overload the save tool `wsave` as discussed in [Saving Tools](@ref) to handle your plot data types. Most likely you will only get [`safesave`](@ref) automatically. Here is an example:
+
+!!! note
+    The example below assumes your using the [Plots.jl](https://github.com/JuliaPlots/Plots.jl) package and for the most part should work with any backend. The `Plots.savefig` function handles the image type based on the extension used, for example, `savefig(example,"example.png")`
+
+```julia
+using DrWatson
+using Plots
+DrWatson._wsave(filename,plot::T) where T<: Plots.Plot = begin
+    mkpath(dirname(filename));
+    savefig(plot,filename)
+end
+```
+The `filename` can be provided by [`savename`](@ref) to assist with adding parameters used to generate the data returned by [`collect_results`](@ref). Nothing new really here, but this raises the question, what if we want to get information from [`tagsave`](@ref) which provides critical details from `git` and the macro [`@tagsave`](@ref) gives us information about the script that ran and line number. We need to then work with text based plots.
+
+### Adding provenance details to text based plots
+ There are many instances where you will want to know what script generated a plot. This is hard to do consistently with plots stored in formats like `pdf` or `png`, but text-based vector formats like [`PGF/tikz`](https://en.wikipedia.org/wiki/PGF/TikZ) (or [`svg`](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics)) are just mark-up text and allow for in-file comments. Therefore, one can store comments in the actual plot file providing provenance information which enables reproducing plots/figures more easily. To do this in `DrWatson`, we have to create a new [`tagsave`](@ref) method for our data type and `Dict` of `info` we want to write. Below is an example:
```

Comment:
I recently had the need to do a similar thing for PDF files, so if there's any interest in including this, the function is actually not too hard with `Ghostscript_jll`:

```julia
using Ghostscript_jll
const AUTHOR = Ref{String}("Anshul Singhvi")

function insert_pdf_metadata(
    file, desc::String;
    title = splitext(splitdir(path)[2])[1],
    author = AUTHOR[],
    keywords = [],
    additional_gs_flags = String[]
    )

    path = abspath(file)
    @assert endswith(path, ".pdf")
    @assert isfile(path)

    gs_options = Cmd(vcat(additional_gs_flags, [
        "-dBATCH",                      # run ghostscript in batch mode
        "-dNOPAUSE",                    # run without pause, exit immediately after complete
        "-dQUIET",                      # no output to stdout
        "-dProcessDSCComments=false",   # try to maintain metadata (this fails) see https://bugs.ghostscript.com/show_bug.cgi?id=693400
        "-sDEVICE=pdfwrite",            # use the `pdfwrite` device
        "-sOutputFile=-",               # output to stdout
    ]))

    gs_cmds = ``` $gs_options -f $path -c "[ /Subject ($desc) /Keywords ($(join(keywords, ", "))) /DOCINFO pdfmark"```

    pdf = Ghostscript_jll.gs() do gs_exe
        read(`$gs_exe $gs_cmds`, String)
    end

    write(path, pdf)

    return true

end
```

You can invoke the function like so:
```julia
path = "/path/to/pdf.pdf"
using Dates # just for the example, not required
insert_pdf_metadata(path, "This was written by Ghostscript!"; keywords = ["a = 1", "b = 1234.8403", "run 1 $(Dates.today())"])
```
The metadata displayed is:
<img width="366" alt="Screen Shot 2022-05-07 at 10 22 03 AM" src="https://user-images.githubusercontent.com/32143268/167258419-0b1704c8-516e-4c74-9fe8-cad026c18fac.png">
<img width="366" alt="Screen Shot 2022-05-07 at 10 22 06 AM" src="https://user-images.githubusercontent.com/32143268/167258421-65a753d8-680a-433e-b115-64a42b0def61.png">